python/tests/test_dataset.py

```diff
@@ -56,54 +56,127 @@ def test_del_item_missing():
         del d['not an item']
 
 
-def test_set_coord():
-    d = sc.Dataset()
-    d.set_coord(Dim.X, sc.Variable(1.0))
-    assert len(d) == 0
+def test_coord_setitem():
+    var = sc.Variable([Dim.X], values=np.arange(4))
+    d = sc.Dataset({'a': var}, coords={Dim.X: var})
+    with pytest.raises(RuntimeError):
```

Comment:
Could match the exact error message?

Reply:
I had previously asserted on exception texts, and it just created a maintenance overhead, since the texts kept changing slightly. On the C++ side I have thus started adding more exception types, which are more stable. We still need to look into what needs to be done in pybind11 to have those available in Python.